AppController/djinn.rb

```diff
@@ -1529,8 +1517,17 @@ def reserve_app_id(username, app_id, app_language, secret)
   #   secret: Shared key for authentication
   #
   def stop_app(app_name, secret)
-    if !valid_secret?(secret)
-      return BAD_SECRET_MSG
+    return BAD_SECRET_MSG if !valid_secret?(secret)
+
+    if !my_node.is_shadow?
+      Djinn.log_debug("Sending stop_app call for #{app_name} to shadow.")
+      acc = AppControllerClient.new(get_shadow.private_ip, @@secret)
+      begin
+        return acc.stop_app(app_name)
+      rescue FailedNodeException => except
+        Djinn.log_warn("Failed to forward stop_app call to shadow (#{get_shadow}).")
+        return NOT_READY
```

Comment:
No: if we are not able to talk to the leader, we are either isolated or the deploymnet is in a flux. We should let the user retry. At least in my mind :)